rpcserver.go

```diff
@@ -3769,6 +3769,168 @@ func marshalOutputType(outputType tappsbt.VOutputType) (taprpc.OutputType,
+		Amount:        req.Amount,
+	}, nil
+}
+
+// MarshalSignedIgnore converts a signed ignore entry into its RPC counterpart.
+func MarshalSignedIgnore(signedIgnore universe.SignedIgnoreTuple) (
+	unirpc.SignedIgnoreAssetOutPoint, error) {
+
+	var (
+		ignoreTuple = signedIgnore.IgnoreTuple.Val
+		ignoreSig   = signedIgnore.Sig.Val
+	)
+
+	rpcAssetOutPoint := rpcutils.MarshalAssetOutPoint(ignoreTuple.PrevID)
+
+	return unirpc.SignedIgnoreAssetOutPoint{
+		IgnoreAssetOutpoint: &unirpc.IgnoreEntry{
+			AssetOutPoint: rpcAssetOutPoint,
+			Amount:        ignoreTuple.Amount,
+		},
+		Sig: ignoreSig.Serialize(),
+	}, nil
+}
+
+// IgnoreAssetOutPoint allows an asset issuer to mark a specific asset outpoint
+// as ignored. An ignored outpoint will be included in the next universe
+// commitment transaction that is published.
+func (r *rpcServer) IgnoreAssetOutPoint(ctx context.Context,
+	rpcReq *unirpc.IgnoreAssetOutPointRequest) (
+	*unirpc.IgnoreAssetOutPointResponse, error) {
+
+	// Unmarshal the request into the native type.
+	req, err := unmarshalIgnoreAssetOutPointRequest(rpcReq)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal request: %w", err)
+	}
+
+	assetID := req.AssetOutPoint.ID
+
+	// Fetch the asset group for the given asset ID.
+	assetGroup, err := r.cfg.TapAddrBook.QueryAssetGroup(ctx, assetID)
+	if err != nil {
+		return nil, fmt.Errorf("fail to find asset group given "+
+			"asset ID: %w", err)
+	}
+
+	// Formulate an asset specifier from the asset ID and group key.
+	assetSpec := asset.NewSpecifierOptionalGroupKey(
+		assetID, assetGroup.GroupKey,
+	)
+
+	// Retrieve asset meta reveal for the asset ID. This will be used to
+	// obtain the supply commitment delegation key.
+	metaReveal, err := r.cfg.TapAddrBook.FetchAssetMetaForAsset(
+		ctx, assetID,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("faild to fetch asset meta: %w", err)
+	}
+
+	// Extract supply commitment delegation pub key from the asset metadata.
+	delegationPubKey, err := metaReveal.DelegationKey.UnwrapOrErr(
+		fmt.Errorf("delegation key not found for given asset"),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// Fetch the full delegation key locator.
+	delegationKey, err := r.cfg.TapAddrBook.FetchInternalKeyLocator(
+		ctx, &delegationPubKey,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch delegation key "+
+			"locator: %w", err)
+	}
+
+	// Formulate the ignore entry and sign it with the delegation key.
+	ignoreTuple := universe.IgnoreTuple{
+		PrevID: req.AssetOutPoint,
+		Amount: req.Amount,
+	}
+
+	ignoreSig, err := ignoreTuple.GenSig(
+		ctx, r.cfg.Lnd.Signer, delegationKey,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to sign ignore tuple: %w", err)
+	}
+
+	// Upsert a signed ignore tuple into the ignore tree archive and
+	// get back the authenticated ignore tuples.
+	signedIgnore := universe.NewSignedIgnoreTuple(ignoreTuple, ignoreSig)
+	authIgnoreTuples, err := r.cfg.UniverseArchive.UpsertIgnoreTuples(
```

Comment:
Rather than insert directly into the DB, we should call into the supply commit state machine: https://github.com/lightninglabs/taproot-assets/blob/648cb72d8b3de9d3f3324ee24b777f918121693a/universe/supplycommit/supply_commit_states.go#L96-L101